src/core/syscalls/syscall_handler.rs

```diff
@@ -1430,4 +1432,137 @@ mod tests {
             Ok(&class_hash)
         );
     }
+
+    #[test]
+    fn test_deploy_and_invoke() {
+        /*
+        DEPLOY
+        */
+        let mut vm = vm!();
+        add_segments!(vm, 4);
+
+        // insert data to form the request
+        memory_insert!(
+            vm,
+            [
+                ((1, 0), (2, 0)), //  syscall_ptr
+                ((2, 0), 10),     // DeployRequestStruct._selector
+                // ((2, 1), class_hash),     // DeployRequestStruct.class_hash
+                ((2, 2), 12),     // DeployRequestStruct.contract_address_salt
+                ((2, 3), 1),      // DeployRequestStruct.constructor_calldata_size
+                ((2, 4), (3, 0)), // DeployRequestStruct.constructor_calldata
+                ((2, 5), 0),      // DeployRequestStruct.deploy_from_zero
+                ((3, 0), 250)     // constructor
+            ]
+        );
+
+        let class_hash_felt = Felt::from_str_radix(
+            "284536ad7de8852cc9101133f7f7670834084d568610335c94da1c4d9ce4be6",
+            16,
+        )
+        .unwrap();
+        let class_hash: [u8; 32] = class_hash_felt.to_bytes_be().try_into().unwrap();
```

Comment:
the function `felt_to_hash()` may be used in this case to avoid the try_into and the unwrap